### PR TITLE
Fixes for analyze-schema objectname for INDEX, TRIGGER, POLICY and OPERATOR object types

### PR DESCRIPTION
**Pull request by priyanshi-yb**
- Fix `INDEX`,  `TRIGGER`, and `POLICY` object names in the schema summary for unique identification   
https://yugabyte.atlassian.net/browse/DB-12602
- Fixed OPERATOR identifier regex to properly give the object name 
https://yugabyte.atlassian.net/browse/DB-12602
- Fixed the AllLowerCase() of `sqlname` class to consider the special chars (`$,%,#, etc..`) and `_` properly
       -  Added unit test for this

Jenkins - 
1. PG complex schemas https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/3744
2. PG long running complex schemas https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/3735

### CODE REVIEW

File: yb-voyager/src/utils/sqlname/sqlname.go [hunk 1]
```diff
@@ -239,7 +240,11 @@ func IsAllUppercase(s string) bool {
 
 func IsAllLowercase(s string) bool {
 	for _, c := range s {
-		if c >= 'A' && c <= 'Z' {
+		if c == '_' {
+			continue
+		}
+		if (c >= 'A' && c <= 'Z') ||
```
Comment (makalaaneesh) [inline yb-voyager/src/utils/sqlname/sqlname.go:246]: The OR is a bit confusing IMO. 
How about we check if above this line if `IsLetter` or `isDigit` and then just continue?

File: yb-voyager/src/utils/sqlname/sqlname.go [hunk 2]
```diff
@@ -239,7 +240,11 @@ func IsAllUppercase(s string) bool {
...
+			!(unicode.IsLetter(rune(c)) || unicode.IsDigit(rune(c))) { // check for special chars
```
Comment (makalaaneesh) [inline yb-voyager/src/utils/sqlname/sqlname.go:247]: Looks like unicode pkg also has an IsUpper? See if we can directly use that? https://stackoverflow.com/a/59293875

Another (non-efficient) solution I like is this :) https://stackoverflow.com/a/59293612

File: yb-voyager/cmd/analyzeSchema.go [hunk 3]
```diff
@@ -62,6 +63,7 @@ var (
 	optionalWS = `[\s\n\t]*` //optional white spaces
 	//TODO: fix this ident regex for the proper PG identifiers syntax - refer: https://github.com/yugabyte/yb-voyager/pull/1547#discussion_r1629282309
 	ident                   = `[a-zA-Z0-9_."-]+`
+	operatorIdent           = `[a-zA-Z0-9_."-+*/<>=~!@#%^&|` + "`" + `]+` // refer https://www.postgresql.org/docs/current/sql-createoperator.html
```
[the remark this answers is not in the document]
Comment (priyanshi-yb) [reply yb-voyager/cmd/analyzeSchema.go:66]: I was thinking about that but thought, I would need to handle it differently there but I think yeah can use parse let me try that.

File: yb-voyager/cmd/analyzeSchema.go [hunk 4]
```diff
@@ -1227,6 +1229,7 @@ func checkSql(sqlInfoArr []sqlInfo, fpath string) {
 			reportCase(fpath, "REFERENCING clause (transition tables) not supported yet.",
 				"https://github.com/YugaByte/yugabyte-db/issues/1668", "", "TRIGGER", trig[1], sqlInfo.formattedStmt, UNSUPPORTED_FEATURES, "")
 		} else if trig := constrTrgRegex.FindStringSubmatch(sqlInfo.stmt); trig != nil {
+			summaryMap["TRIGGER"].invalidCount[fmt.Sprintf("%s ON %s", trig[1], trig[3])] = true
```
[the remark this answers is not in the document]
Comment (priyanshi-yb) [reply yb-voyager/cmd/analyzeSchema.go:1232]: > the summaryMap for, is it counts?

yes it for summary, so counts, object names etc..

> Do you still need this given what we're doing
No for this case we don't need it as we don't handle CONSTRAINT trigger in the summary itself, this was added by-mistake as I trying to add support for that in regex of CREATE TRIGGER but I was facing some other issues with that. Will remove this.

File: yb-voyager/cmd/analyzeSchema.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (makalaaneesh) [reply yb-voyager/cmd/analyzeSchema.go:66]: Okay to do it in a separate PR

File: yb-voyager/cmd/analyzeSchema.go [hunk 5]
```diff
@@ -1511,6 +1521,41 @@ func processCollectedSql(fpath string, stmt string, formattedStmt string, objTyp
 	return sqlInfo
 }
 
+func getObjectNameWithTable(stmt string, regexObjName string) string {
+	parsedTree, err := pg_query.Parse(stmt)
+	if err != nil {
+		// in case it is not able to parse stmt as its not in PG syntax so returning the regex name
```
Comment (makalaaneesh) [inline yb-voyager/cmd/analyzeSchema.go:1526]: log the error

File: yb-voyager/cmd/analyzeSchema.go [hunk 6]
```diff
@@ -1467,7 +1470,14 @@ func processCollectedSql(fpath string, stmt string, formattedStmt string, objTyp
 	//update about sqlStmt in the summary variable for the report generation part
 	createObjStmt := createObjRegex.FindStringSubmatch(formattedStmt)
 	if createObjStmt != nil {
-		objName = createObjStmt[objNameIndex]
+		if slices.Contains([]string{"INDEX", "POLICY", "TRIGGER"}, objType) {
```
Comment (makalaaneesh) [inline yb-voyager/cmd/analyzeSchema.go:1472]: Could there be other types that could require same treatment? Just do a scan once ..

**Review by makalaaneesh — COMMENTED** [review]
LGTM

File: yb-voyager/cmd/analyzeSchema.go (hunk 6 above)
Comment (priyanshi-yb) [reply yb-voyager/cmd/analyzeSchema.go:1472]: I don't think there can be anything other than these.

**Review by makalaaneesh — APPROVED** [review]

**Pull request merged**